doc/adrs/086_provide_the_variables_available_for_each_operation.adoc

```diff
@@ -0,0 +1,170 @@
+
+  @Override
+  public List<Variable> getVariables(String operation) {
+    return switch (operation) {
+      case "Diagram#canCreate", "Diagram#label", "Diagram#targetObjectId" -> List.of(ENVIRONMENT, EDITING_CONTEXT, SELF);
+      case "Node#semanticElements" -> List.of(ENVIRONMENT, EDITING_CONTEXT, SELF);
+      case "Edge#shouldRender" -> List.of(ENVIRONMENT, EDITING_CONTEXT, SELF, SEMANTIC_EDGE_SOURCE, SEMANTIC_EDGE_TARGET, GRAPHICAL_EDGE_SOURCE, GRAPHICAL_EDGE_TARGET)
+      default -> List.of();
+    }
+  }
+}
+----
+
+=== Attach operations
+
+After that, we will attach the operation definition to the code responsible for their execution.
+For that, we can use a simple annotation to indicate that some Java handler is used to implement a specific operation:
+
+[source,java]
+----
+@Operation("Node#semanticElements")
+public class SemanticCandidatesProvider implements Function<VariableManager, List<?>> {
+  public List<?> apply(VariableManager variableManager) {
+    return List.<Object>of();
+  }
+}
+----
+
+Using this annotation, we could start validating the variables attached to the operation to ensure that our expectation are matching what is really happening:
+
+[source,java]
+----
+public interface IOperationValidator {
+  void validate(Object object, VariableManager variableManager);
+}
+----
+
+[source,java]
+----
+@Operation("Node#semanticElements")
+public class SemanticCandidatesProvider implements Function<VariableManager, List<?>> {
+  private final IOperationValidator operationValidator;
+
+  public List<?> apply(VariableManager variableManager) {
+    this.operationValidator.validate(this, variableManager);
+
+    return List.<Object>of();
+  }
+}
+----
+
+The operation validator would have for only job to check the `@Operation` annotation on the class and ensure that the variables found at runtime are matching what was expected.
+It could thus log as a warning missing variables and as an info additional variables.
+
+In a similar fashion, we can attach the operation to an expression using an EAnnotation on object of the View DSL to indicate that the feature NodeDescription#semanticCandidatesExpression is linked to the operation `Node#semanticElements`.
+
+
+=== Generate the documentation
+
+It should be possible after that to generate the list of all the available variables for each expression of the view DSL and for all Java handlers.
+
+== Status
+
+WIP
+
+== Consequences
+
+Such support would give us the ability to list all the variables which should be available for all operations of your representations both in Java and in the View DSL.
+This would represent a foundation which would be improved in the future to solve additional problems such as:
+
+- Dynamically compute the type of the variables (specifying a domain type in the View DSL should change the type of `self`)
+- Dynamically compute the list of the variables using the description
+- Define the expected result for every operations
```

Comment:
Also:
- support more precise completion of AQL expressions.

Reply:
Done